File: content/examples/other/leetcode/problems/implement-trie/example-1/solution.py
class TrieSimple:
    def __init__(self):
        self.words = set()

    def insert(self, word):
        self.words.add(word)

    def search(self, word):
        return word in self.words

    def starts_with(self, prefix):
        for w in self.words:
            if w.startswith(prefix):
                return True
        return False


if __name__ == "__main__":
    trie = TrieSimple()
    trie.insert("apple")
    print(trie.search("apple"))
    print(trie.search("app"))
    print(trie.starts_with("app"))

File: content/examples/other/leetcode/problems/implement-trie/example-1/solution.js
class TrieSimple {
  constructor() {
    this.words = new Set();
  }

  insert(word) {
    this.words.add(word);
  }

  search(word) {
    return this.words.has(word);
  }

  startsWith(prefix) {
    for (const w of this.words) {
      if (w.startsWith(prefix)) return true;
    }
    return false;
  }
}

if (require.main === module) {
  const trie = new TrieSimple();
  trie.insert("apple");
  console.log(trie.search("apple"));
  console.log(trie.search("app"));
  console.log(trie.startsWith("app"));
}

module.exports = { TrieSimple };
